Create lambda template for dr2 lambdas.

MancunianSam opened the pull request:
Description of files and inputs is in the README.

techncl commented on src/main/g8/default.properties:
```diff
@@ -0,0 +1,5 @@
+name =
```
Could this be renamed to something like "repo name", "project name" etc just so silly people like me (...hypothetically speaking of course 👀 ) don't think it's their own name that they should put?

techncl reviewed (commented):
Looks good (and it's a really good idea as it's easy to forget certain files); I added some comments throughout and left some general ones here:

1. Should we add a blank `application.conf`?
2. Should there be an "Are you happy with this configuration?" prompt before it gets generated?
3. Would be possible (good idea) to add a githook so that when you commit, it does a `scalafmtCheckAll`?

MancunianSam commented on src/main/g8/default.properties:
```diff
@@ -0,0 +1,5 @@
+name =
```
It can't be renamed, it's a reserved word in the giter8 template which determines what the folder is called. 

We can definitely add a full_name variable in. I've set it to default to test-repo -> Test Repo but that can then be overridden.

kurtismash replied:
Do we want our lambda repos to end `-lambda` so it's clear that the repo is for a lambda?

MancunianSam replied:
I don't know. It's possible that the code could be shifted to run somewhere other than lambda in the future and then that's just confusing. It shouldn't really matter reading the code where it runs. What do you think @techncl ?

MancunianSam replied:
Also, because the reserved `name` is used for the folder name, I don't think you can add a suffix to it like that.

techncl replied:
It would make it easier when searching through Github but then again, in point out that it's a Lambda, it could/should be made clear in the "description" (which is also searchable). Maybe instead of an empty string as the default for the "description", we could add "Lambda" or "A Lambda" to encourage people to complete the description?

MancunianSam replied:
That makes sense, I've added in a default description of "A Lambda"

MancunianSam commented on src/main/g8/.gitignore:
```diff
@@ -0,0 +1,3 @@
+.idea/
+target/
+.bsp/
```
Good idea, I've added it in.

MancunianSam commented on src/main/g8/src/main/scala/$organisation__packaged$/Lambda.scala:
```diff
@@ -0,0 +1,17 @@
+package $organisation$
+
+$if(use_input_stream.truthy)$
```
Sadly, you can't. The CLI input options are pretty limited. If you do put both, you'll get a nonsense Scala class which won't compile which will encourage people to read the instructions :smile:

MancunianSam commented on README.md:
```diff
@@ -1 +1,50 @@
+Kebab case should be used.
+### organisation 
+Used for the package structure and the organisation in build.sbt
+### use_sqs_event
```
It does show it's the default. On the prompt it says TRUE/false or true/FALSE for the opposite. The CLI options are pretty limited.

MancunianSam commented:
I did do an `application.properties` but `application.conf` makes more sense, I've changed it.

You can't get an "are you happy prompt", there aren't a lot of options for the CLI.

I've added in a config file for pre-commit. It's also what TRE are using so that lines up. This runs scalaFmt on commit. There's no way I think to get this installed automatically when the template is created so I've added instructions in the README to show how to set it up.

techncl commented on README.md:
```diff
@@ -1 +1,50 @@
+Used for the package structure and the organisation in build.sbt
+### use_sqs_event
+Creates a lambda which takes an SQS event. Can't be used with use_input_stream
+### use_input_stream
```
Should we maybe change it to `use_event`? And then in the README add something like

"Creates a lambda which takes an event **(SQS event but default but you can easily change this).** Can't be used with use_input_stream"?

MancunianSam replied:
Yeah, that makes a lot of sense. I've changed it.

techncl reviewed (approved):
Looks good
